client-js/app/elements/labrad-registry.ts

```diff
@@ -387,25 +395,40 @@ export class LabradRegistry extends polymer.Base {
       try {
         this.$.pendingDialog.open();
         this.$.pendingOp.innerText = "Copying...";
-        var resp = await this.socket.copyDir({path: oldPath, dir: oldName, newPath: newPath, newDir: newName});
-        this.$.toastCopySuccess.show();
+        switch (this.$.dragClass.innerText) {
```

Comment:
Agreed on both fronts. Adding dialogs involves a lot more boilerplate than necesary. I'll fix this using an attribute.